### PR TITLE
docs for instance scoped system variables

### PR DESCRIPTION
**Pull request by morgo**
### What is changed, added or deleted? (Required)

This adds the documentation changes described earlier in https://github.com/pingcap/tidb/blob/master/docs/design/2021-12-08-instance-scope.md#documentation-changes

In master (and 6.0) we have native support for instance scoped variables, which are now set with "SET GLOBAL" instead of "SET SESSION" (note: `SET INSTANCE` was never a thing.)

We don't describe instance scoped variables as "instance scoped" but rather: "Persists to cluster: No". This was intentional because INSTANCE isn't really a thing in MySQL. The property that is most understantable is if it persists to the other nodes in the cluster, and INSTANCE and GLOBAL scope are mutually exclusive.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/32888

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md
Comment (morgo) [inline system-variables.md:954]: This is an additional change that was picked up by the script. The default has changed in TiDB 6.0.

**Comment by CbcWestwolf** [comment]
LGTM

**Review by dveeden — APPROVED** [review]

**Comment by dveeden** [comment]
Might be good to check after merging if any other PRs have updated this file and make sure any newly added config options also have the right information. Maybe also check for other open PRs for the same file.

**Comment by morgo** [comment]
> Might be good to check after merging if any other PRs have updated this file and make sure any newly added config options also have the right information. Maybe also check for other open PRs for the same file.

Currently it's easier to catch it on the next auto-generated config PR. This is generated from https://github.com/pingcap/docs/pull/5720 - I would like to add it to CI at some point, but we still have 70 sysvars to document first :(

File: system-variables.md
Comment (morgo) [inline system-variables.md:252]: This is from https://github.com/pingcap/tidb/pull/33473 - it only affects master / TiDB 6.1.

File: system-variables.md
Comment (morgo) [inline system-variables.md:517]: This is from https://github.com/pingcap/tidb/pull/30664 - it only affects master and not TIDB 6.0.

File: system-variables.md
Comment (morgo) [inline system-variables.md:825]: This was added to 6.0 but not documented. It is the new feature being documented in this PR. @CbcWestwolf will translate this to Chinese.

File: system-variables.md
Comment (morgo) [inline system-variables.md:842]: The default is wrong. It is OFF not on. See https://github.com/pingcap/tidb/pull/31547 by @[ekexium](https://github.com/ekexium)

(It is wrong in 6.0 and master)

File: system-variables.md
Comment (morgo) [inline system-variables.md:907]: These have just been moved. They were not in alphabetical order :(

File: system-variables.md
Comment (morgo) [inline system-variables.md:1140]: The documentation for Instance scope variables changes to "GLOBAL + Persists = No". See: https://github.com/pingcap/tidb/blob/master/docs/design/2021-12-08-instance-scope.md

File: system-variables.md
Comment (morgo) [inline system-variables.md:1607]: This is a bug fix in https://github.com/pingcap/tidb/issues/32763 - it is function neutral.

File: system-variables.md
Comment (TomShawn) [inline system-variables.md:1687]: `(New in v6.0.0)` is a makeshift format. If a heading with `<span class="version-mark">` is referenced in an anchor, even if it passes the CI link check, the <docs.pingcap.com> site no longer parses the anchor as it always did, probably by a frontend misoperation. Maybe they will fix it in the future. So `(New in v6.0.0)` is a makeshift format that makes sure the anchor works and also makes the CI happy.

Comment (TomShawn) [reply system-variables.md:1687]: If we write the heading as `### tidb_sysdate_is_now <span class="version-mark">New in v6.0.0</span>` as we did before, and reference it in an anchor `[](/system-variables.md#tidb_sysdate_is_now-new-in-v600)`, when the docs site builds is ready and you click on the referenced anchor link, you **will not** be directed to the exact heading position.

Comment (morgo) [reply system-variables.md:1687]: It is inconsistent with the previous format though (and looks weird).

I can change them all to be in the format:

```
### VariableName
- Introduced: {Version Number}
- Default value: `0`
- Range: `[x, y]`
```

This would make the anchors more stable, but I would suggest we do it in a followup PR because they'll all change.

Comment (TomShawn) [reply system-variables.md:1687]: Good idea!

**Review by TomShawn — APPROVED** [review]
LGTM

File: system-variables.md
Comment (TomShawn) [inline system-variables.md:28]:
```suggestion
>
> - For variables with the `Persists to cluster: Yes` setting, when a global variable is changed, a notification is sent to all TiDB servers to refresh their system variable cache. When you add additional TiDB servers or restart existing TiDB servers, the persisted configuration value is automatically used.
```

**Comment by TomShawn** [comment]
@ran-huang Shall we merge this PR now? All contents are applicable to v6.1.

**Comment by morgo** [comment]
> @ran-huang Shall we merge this PR now? All contents are applicable to v6.1.

If you don't merge it, followup PRs will be more complicated because more is picked up by the script.

**Comment by ran-huang** [comment]
> > @ran-huang Shall we merge this PR now? All contents are applicable to v6.1.
> 
> If you don't merge it, followup PRs will be more complicated because more is picked up by the script.

If this PR is **reviewed and approved by the code owner(s)**, we can merge it. 

I understand that the follow-up PRs will be more complicated, and we hope to merge every PR as soon as we can. I only have one concern: these changes are applicable to v6.1, but since v6.1 has not been released yet, there might be reverts or further changes to the code -- it happened a lot in the past few releases. If that happens, I'm afraid no one will remember to make changes to the docs accordingly. That's the major reason we insist that this PR be merged after code freeze. 

If we merge this PR before code freeze, we'll need to get back to it and confirm there are no further changes. Can we ask for your help by then?

**Comment by morgo** [comment]
> If we merge this PR before code freeze, we'll need to get back to it and confirm there are no further changes. Can we ask for your help by then?

Sure that's fine - these changes are generated by a script, so if they are reverted (which is rare) a subsequent script execution will update to the new values.

But I have to say that its rare we revert PRs - and a more likely outcome of this process is that it is leading to the system variables docs being out of date. All but 2 changes also apply to 6.0, but because the last PR to update the docs took too long to review ( https://github.com/pingcap/docs/pull/7737) we had to delay this to now for documenting (the changes stack on each other, and get complicated to review if we don't merge them).

**Review by bb7133 — APPROVED** [review]
LGTM

**Comment by bb7133** [comment]
> But I have to say that its rare we revert PR

I agree with it.

**Comment by yudongusa** [comment]
LGTM

**Pull request merged**
